### DISCUSSION TITLE
setControlModes

### DISCUSSION THREAD

**Discussion by jimingda** [pattacini/study-air-hockey#7] [Q&A] [answered]
In study-2, why didn't set the following before `ipos[i]->positionMove(poss.data());` , but after the `positionMove`
```
        std::fill(begin(modes), end(modes), VOCAB_CM_POSITION_DIRECT);
        imod->setControlModes(modes.data());
```
In my understanding, the setControlModes should be set before positionMove.

**Comment by pattacini** [comment, marked as answer]
Hi @jimingda 

The method `setControlModes()` aims to switch the control mode for a set of joints. The two control modes in question are:
- `VOCAB_CM_POSITION`
- `VOCAB_CM_POSITION_DIRECT`

In fact, `setControlModes()` gets called twice within the method `helperOpenDevice()`. In detail:

The lines below allow for switching to standard position control to let the robot reach its starting configuration.

https://github.com/pattacini/study-air-hockey/blob/35dbfff2d7636fa80f1f6c2a083ff06142cd2d7d/src/study-2.cpp#L71-L100

Once done, these lines enable the position-direct mode that will be used later on by the code of the `updateModule()`:

https://github.com/pattacini/study-air-hockey/blob/35dbfff2d7636fa80f1f6c2a083ff06142cd2d7d/src/study-2.cpp#L102-L103

Hope this clarifies the matter.
